lib: Update identity_key and hw_uniq_key libraries to return error codes (#11557)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

  * :ref:`nrf_security` library, relocated from the sdk-nrfxlib repository to the :file:`subsys/nrf_security` directory.

> * :ref:`lib_hw_unique_key` library:

Reviewer: Move both lists to Other libraries section.

Author: Done

  * Updated:

    * :c:func:`hw_unique_key_write`, :c:func:`hw_unique_key_write_random` and :c:func:`hw_unique_key_load_kdr` functions updated to return an error code and not panic on error.
    * :c:func:`hw_unique_key_derive_key` function is updated to always return an error code from the library defined codes and not forward error codes from underlying libraries.

* :ref:`lib_identity_key`

  * Updated:

>     * :c:func:`identity_key_write_random`, :c:func:`identity_key_write_key` and :c:func:`identity_key_write_dummy` functions updated to return an error code and not panic on error.
>     * :c:func:`identity_key_read` function is updated to always return an error code from the library defined codes.

Reviewer: ```suggestion
    * :c:func:`identity_key_write_random`, :c:func:`identity_key_write_key` and :c:func:`identity_key_write_dummy` functions to return an error code and not panic on error.
    * :c:func:`identity_key_read` function to always return an error code from the library-defined codes.
```
Compound modifier
library defined codes -> library-defined codes

  * :ref:`nrf_security` library, relocated from the sdk-nrfxlib repository to the :file:`subsys/nrf_security` directory.

* :ref:`lib_hw_unique_key` library:

  * Updated:

    * :c:func:`hw_unique_key_write`, :c:func:`hw_unique_key_write_random` and :c:func:`hw_unique_key_load_kdr` functions to return an error code and not panic on error.
    * :c:func:`hw_unique_key_derive_key` function to always return an error code from the library-defined codes and not forward error codes from underlying libraries.

> * :ref:`lib_identity_key`

Reviewer: Amazingly enough, missed this in the first round :)
```suggestion
* :ref:`lib_identity_key` library:
```

Author: Done

Other libraries
---------------

* :ref:`lib_identity_key` library:

  * Updated:

    * :c:func:`identity_key_write_random`, :c:func:`identity_key_write_key` and :c:func:`identity_key_write_dummy` functions to return an error code and not panic on error.
    * :c:func:`identity_key_read` function to always return an error code from the library-defined codes.

* :ref:`lib_hw_unique_key` library:

  * Updated:

    * :c:func:`hw_unique_key_write`, :c:func:`hw_unique_key_write_random` and :c:func:`hw_unique_key_load_kdr` functions to return an error code and not panic on error.

>     * :c:func:`hw_unique_key_derive_key` function to always return an error code from the library-defined codes and not forward error codes from underlying libraries.

Reviewer 2: Nit: Strange to use different wording with the above, which is the same change. Maybe align them?

Author: Done

Other libraries
---------------

* :ref:`lib_identity_key` library:

  * Updated:

    * :c:func:`identity_key_write_random`, :c:func:`identity_key_write_key` and :c:func:`identity_key_write_dummy` functions to return an error code and not panic on error.
    * :c:func:`identity_key_read` function to always return an error code from the library-defined codes.

>     * Updated the error code defined names to follow the prefix IDENTITY_KEY_ERR_*.

Reviewer: ```suggestion
    * The defined error code names with prefix IDENTITY_KEY_ERR_*.
```

* :ref:`lib_hw_unique_key` library:

  * Updated:

    * :c:func:`hw_unique_key_write`, :c:func:`hw_unique_key_write_random` and :c:func:`hw_unique_key_load_kdr` functions to return an error code and not panic on error.
    * :c:func:`hw_unique_key_derive_key` function to always return an error code from the library-defined codes.

>     * Updated the error code defined names to follow the prefix HW_UNIQUE_KEY_ERR_*.

Reviewer: ```suggestion
    * The defined error code names with prefix HW_UNIQUE_KEY_ERR_*.
```
